app/models/post.rb

```diff
@@ -3,11 +3,39 @@
 class Post < ActiveRecord::Base
   include Taggable
 
-  include Elasticsearch::Model
+  # Elasticsearch::Model doesn't work well with STI, so
+  # include it in subclasses directly.
+  def self.inherited(child)
+    super
+
+    child.instance_eval do
+      include Elasticsearch::Model
+
+      after_commit :index_document, on: :create
+      after_commit :update_or_delete_document, on: :update
+      after_commit :delete_document, on: :destroy
+
+      settings(
+        analysis: {
+          analyzer: {
+            normal: {
+              tokenizer: 'standard',
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.